src/compute/src/sink/copy_to_s3_oneshot.rs

```diff
@@ -49,13 +50,43 @@ where
             response_buffer: Some(Rc::clone(&compute_state.copy_to_response_buffer)),
         })));
         let response_protocol_weak = Rc::downgrade(&response_protocol_handle);
+        let connection_context = compute_state.context.connection_context.clone();
 
-        copy_to(
-            sinked_collection,
-            err_collection,
-            sink_id,
+        let callback = move |count: Result<u64, String>| {
+            if let Some(response_protocol) = response_protocol_handle.borrow_mut().deref_mut() {
+                response_protocol.send(count);
+            }
+        };
+
+        // Move all the data to a single worker and consolidate
+        // TODO: split the data among worker to some number of buckets (either static or configurable
+        // by the user) to split the load across the cluster.
```

Comment:
Nit:

```suggestion
        // TODO(#7256): split the data among worker to some number of buckets (either static or configurable
        // by the user) to split the load across the cluster.
```

To ensure this TODO gets resolved before the Epic is closed.